### PR TITLE
Punchier tera animation

### PR DESCRIPTION
**Pull request by Pokabbie**
## Description

Adds a punchier animation for terastalizing.

I'm not particularly good at animation, so it basically just mimics the Mega Evolution animation with some different graphics and sound effects.

## Images
Before
tera_before

After
tera_after

## **Discord contact info**
@Pokabbie

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
> If we have a better animation, I don't think we should have the text anymore?

I think it's fine, for clarity.

**Comment by Pokabbie** [comment]
I found in a few playtests of my hack, people got confused by which type it was as some of the UI indicators aren't immediately clear (Especially to players who aren't as big into Pokémon).
So, unless the tera hats are somehow added, keeping the text feels like a necessity imo

File: asm/macros/battle_script.inc [hunk 1]
```diff
@@ -2277,6 +2277,10 @@
 	various \battler, VARIOUS_ACTIVATE_TERRAIN_CHANGE_ABILITIES
 	.endm
 
+	.macro applyterastallization battler:req
+	various \battler, VARIOUS_APPLY_TERASTALLIZATION
+	.endm
+
```
Comment (AsparagusEduardo) [inline asm/macros/battle_script.inc:2283]: Can you move this to a callnative? We're slowly deprecating the use of `various`. An example would be `BS_HandleMegaEvolution` for example.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Just a small tweak needed imo. Looks good otherwise 👍

File: asm/macros/battle_script.inc (hunk 1 above)
Comment (Pokabbie) [reply asm/macros/battle_script.inc]: Done in my latest commit 🙂

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -16919,3 +16912,12 @@ void BS_RemoveWeather(void)
     RemoveAllWeather();
     gBattlescriptCurrInstr = cmd->nextInstr;
 }
+
+void BS_ApplyTerastallization(void)
+{
+    NATIVE_ARGS(u8 battler);
```
Comment (AlexOn1ine) [inline src/battle_script_commands.c:16918]: You don't need to pass the battler argument. You can just use `gBattlerAttacker`

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -16919,3 +16912,12 @@ void BS_RemoveWeather(void)
...
+
+    u8 battler = GetBattlerForBattleScript(cmd->battler);
+    ApplyBattlerVisualsForTeraAnim(battler);
+    gBattlescriptCurrInstr = cmd->nextInstr;
+}
```
Comment (AlexOn1ine) [inline src/battle_script_commands.c:16923]: missing new line

File: src/battle_script_commands.c (hunk 2 above)
Comment (AlexOn1ine) [reply src/battle_script_commands.c:16918]: Yeah, it should could be removed there as well :sweat_smile:

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**